tests/Emitter/SapiEmitterTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Tests\Emitter;
+
+use Nyholm\Psr7\Response;
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Yii\Web\Emitter\SapiEmitter;
+
+/**
+ * @runTestsInSeparateProcesses
+ */
+class SapiEmitterTest extends TestCase
+{
+    public function testEmit()
+    {
+        $body = 'Example body';
+        $response = new Response(200, ['X-Test' => 1], $body);
+
+        ob_start();
+        (new SapiEmitter())->emit($response);
+        $result = ob_get_contents();
+        ob_end_clean();
+
+        $this->assertEquals(200, http_response_code());
+        $this->checkHeadersEquals([
+            'X-Test: 1',
+            'Content-Length: ' . strlen($body),
+        ]);
+        $this->assertEquals($body, $result);
+    }
+
+    public function testEmptyBodyOnCode()
+    {
+        $response = new Response(204, ['X-Test' => 1], 'Example body');
+
+        ob_start();
+        (new SapiEmitter())->emit($response);
+        $result = ob_get_contents();
+        ob_end_clean();
+
+        $this->assertEquals(204, http_response_code());
+        $this->checkHeadersEquals([
+            'X-Test: 1',
+        ]);
+        $this->assertEmpty($result);
+    }
+
+    public function testEmptyBodyOnFlag()
+    {
+        $response = new Response(200, ['X-Test' => 1], 'Example body');
+
+        ob_start();
+        (new SapiEmitter())->emit($response, true);
+        $result = ob_get_contents();
+        ob_end_clean();
+
+        $this->assertEquals(200, http_response_code());
+        $this->checkHeadersEquals([
+            'X-Test: 1',
+        ]);
+        $this->assertEmpty($result);
+    }
+
+    public function testContentLengthHeader()
```

Comment:
What does it test compared to testEmit()?

Reply:
Check what emit() doesn't compute `Content-length` header: 
https://github.com/yiisoft/yii-web/blob/5db3424fefb3a377da1d74ceae80916691b1ac5e/src/Emitter/SapiEmitter.php#L40

Reply:
```suggestion
    /*
     * @test
     */
    public function contentLengthShouldNotBeOverwrittenIfPresent()
```